Merge from main

Anindyadeep opened the pull request.

Janaka-Steph reviewed (approved):
Maybe needs more tests but LGTM otherwise

Anindyadeep commented:
> Maybe needs more tests but LGTM otherwise

Actually most of the tests are API dependent, and I saw some examples by other providers and saw that they did only these tests. 


If everything seems okay, then should I merge this and then put a PR from `prem-main` -> `llama-index:main`?

filopedraz commented:
We can merge and then create PR for llama index.

jerryjliu committed:
doc nit: rearrange module guide  (#12741)

alexsherstinsky committed:
[BUGFIX] Update LlamaIndex-Predibase Integration (#12736)

billytrend-cohere committed:
Update cohere and add async support (#12705)

billytrend-cohere committed:
Revert _ (#12746)

jerryjliu committed:
fix multidoc hyde notebook  (#12742)

hassiebp committed:
fix(docs): langfuse link to notebook on observability page (#12747)

jerryjliu committed:
nit: custom agent pass kwargs (#12739)

rozerarenu committed:
Add Amazon Bedrock knowledge base integration as retriever (#12737)

Javtor committed:
Use None as the default value for s3_endpoint_url in S3Reader (#12763)

use none as default

logan-markewich committed:
Chain-of-Abstraction Agent Pack (#12757)

dependabot[bot] committed:
Bump idna from 3.6 to 3.7 in /llama-index-integrations/storage/index_store/llama-index-storage-index-store-elasticsearch (#12784)

dependabot[bot] committed:
Bump idna from 3.6 to 3.7 in /llama-index-integrations/storage/docstore/llama-index-storage-docstore-elasticsearch (#12783)

dependabot[bot] committed:
Bump idna from 3.6 to 3.7 in /llama-index-networks/examples/privacy_safe_retrieval/contributor-2 (#12782)

dependabot[bot] committed:
Bump idna from 3.6 to 3.7 in /llama-index-networks/examples/privacy_safe_retrieval/contributor-1 (#12781)

dependabot[bot] committed:
Bump idna from 3.6 to 3.7 in /llama-index-integrations/llms/llama-index-llms-llamafile (#12780)

dependabot[bot] committed:
Bump idna from 3.6 to 3.7 in /llama-index-integrations/tools/llama-index-tools-passio-nutrition-ai (#12779)

dependabot[bot] committed:
Bump idna from 3.6 to 3.7 in /llama-index-integrations/llms/llama-index-llms-friendli (#12778)

dependabot[bot] committed:
Bump idna from 3.6 to 3.7 in /llama-index-networks/examples/demo/contributor-1 (#12777)

dependabot[bot] committed:
Bump idna from 3.6 to 3.7 in /llama-index-networks/examples/demo/contributor-2 (#12776)

dependabot[bot] committed:
Bump idna from 3.6 to 3.7 in /llama-index-networks/examples/demo/contributor-3 (#12775)

dependabot[bot] committed:
Bump idna from 3.6 to 3.7 in /llama-index-integrations/tools/llama-index-tools-finance (#12774)

dependabot[bot] committed:
Bump idna from 3.6 to 3.7 in /llama-index-core (#12770)

jerryjliu committed:
nit: COA colab badge  (#12800)

dependabot[bot] committed:
Bump idna from 3.6 to 3.7 in /llama-index-networks (#12769)

dependabot[bot] committed:
Bump idna from 3.6 to 3.7 in /docs (#12764)

dependabot[bot] committed:
Bump idna from 3.6 to 3.7 (#12765)

dependabot[bot] committed:
Bump idna from 3.6 to 3.7 in /llama-index-packs/llama-index-packs-koda-retriever (#12766)

dependabot[bot] committed:
Bump idna from 3.6 to 3.7 in /llama-index-packs/llama-index-packs-diff-private-simple-dataset (#12767)

dependabot[bot] committed:
Bump idna from 3.6 to 3.7 in /llama-index-integrations/agent/llama-index-agent-openai (#12768)

logan-markewich committed:
move pandas to experimental (#12419)

HawkClaws committed:
Add mongodb atlas bm25 retriever (#12519)

Javtor committed:
fix: wrong doc id when using default s3 endpoint in S3Reader (#12803)

* use none as default

* fix doc id

* bump version

alexsherstinsky committed:
[FEATURE] Enable choice of either Predibase-hosted or HuggingFace-hosted fine-tuned adapters in LlamaIndex-Predibase integration. (#12789)

* Enable choice of either Predibase-hosted or HuggingFace-hosted fine-tuned adapters.

* Update the example notebook.

* Updated a docstring.

* Version bump for LlamaIndex-Predibase integration.

logan-markewich committed:
fix message serialization in redis chat store (#12802)

whyiug committed:
fix wrong params (#12773)

fzowl committed:
Update VoyageAI embeddings example (#12752)

logan-markewich committed:
make pip check work for optional pdf packages (#12758)

berrazuriz1 committed:
Fix chunking error (#12715)

* pass bytes instead of text to chunking nodes in CodeHierarchyNodeParser

* fix lint with pre-commit linting

* unit test for special character cases

* vbump

* fix erase white spaces

* fix test name

---------

Co-authored-by: berrazuriz1 <berrazuriz1@uc.cl>
Co-authored-by: Logan Markewich <logan.markewich@live.com>

isaacwasserman committed:
Support for Mistral Large from Bedrock (#12804)

rozerarenu committed:
Update bedrock notebook to use the retriever with bedrock llms (#12771)

hemidactylus committed:
Relax dependency version to accept astrapy 1.* (#12792)

NickSager committed:
Fix bug in raft dataset generator - multiple system prompts (#12751)

nithishr committed:
Add support for Couchbase as a Vector Store (#12680)

atlas-comstock committed:
improvement: convert to Path in file readers if input path str (#12807)

logan-markewich committed:
update async utils (#12812)

yisding committed:
fix pyproject.toml in core for 3.12 (#12762)

logan-markewich committed:
fix for complex conditional links in query pipeline (#12805)

logan-markewich committed:
v0.10.29 (#12817)

timothyckl committed:
Clarify Token Reservation Comment (#12821)

jerryjliu committed:
add OR filter condition support to simple vector store (#12823)

* cr

* cr

* cr

jerryjliu committed:
nits: agent / tool imports, custom agent callbacks (#12824)

zhanghexian committed:
add vearch to vector store (#10972)

Oscilloscope98 committed:
[Community] add `ipex-llm` embedding integration (#12740)

shane-huang committed:
[community] add support for loading "low-bit format" model to `IpexLLM` integration (#12785)

0-yy-0 committed:
fix: pipline load without docstore (#12808)

calebpeffer committed:
feat: Firecrawl Web Loader (#12825)

ashishb committed:
Add the ability to pass custom headers to Anthropic LLM requests (#12819)

rc19 committed:
Use async `_aprocess_actions` in `_arun_step_stream` (#12846)

sumitsahoo committed:
Fix generate_openai_multi_modal_chat_message import in examples doc (#12847)

gblazq committed:
fix: batch insertions into Milvus vector store (#12837)

aman-vink committed:
Update base.py (#12844)

gregory-shklover committed:
Expose azure_ad_token_provider argument to support token expiration (#12818)

xpbowler committed:
Add intermediate outputs to QueryPipeline (#12683)

EmanuelCampos committed:
feat: add retry on pypdf (#12848)

* feat: add retry on pypdf

* chore: correct version

* chore: remove exponential

osman-haider committed:
Find Import Bug (#12861)

ptorru committed:
Octoai embeddings (#12857)

ziodave committed:
see #12874: provide the exception to the StreamChatErrorEvent (#12879)

logan-markewich committed:
Catch validation errors (#12882)

ravi03071991 committed:
Update mistralai api (#12894)

* Update mistralai api

* Update cookbook and function calling model

* Update version

logan-markewich committed:
Fix show progress being out of order (#12897)

billytrend-cohere committed:
Add httpx_async_client option (#12896)

rozerarenu committed:
Add changes for bedrock documentation and notebook (#12856)

logan-markewich committed:
fix bug in load and search tool spec (#12902)

jerryjliu committed:
[WIP] LATS agent pack (#12735)

moria97 committed:
Update pheonix pyproject.toml to support py3.12 (#12862)

logan-markewich committed:
v0.10.30 (#12903)

hanscl committed:
OctoAI Integration (#11375)

Disiok committed:
Fix Bedrock KB retriever (#12910)

wip

ravi03071991 committed:
Add Llama3 cookbook (#12942)

* Add Llama3 cookbook

* Update

* Update

dependabot[bot] committed:
Bump aiohttp from 3.9.3 to 3.9.4 in /llama-index-packs/llama-index-packs-diff-private-simple-dataset (#12940)

dependabot[bot] committed:
Bump aiohttp from 3.9.3 to 3.9.4 in /llama-index-integrations/storage/docstore/llama-index-storage-docstore-elasticsearch (#12939)

dependabot[bot] committed:
Bump aiohttp from 3.9.3 to 3.9.4 in /llama-index-integrations/storage/index_store/llama-index-storage-index-store-elasticsearch (#12938)

dependabot[bot] committed:
Bump aiohttp from 3.9.3 to 3.9.4 in /llama-index-networks/examples/privacy_safe_retrieval/contributor-2 (#12937)

dependabot[bot] committed:
Bump aiohttp from 3.9.3 to 3.9.4 in /llama-index-networks/examples/privacy_safe_retrieval/contributor-1 (#12936)

dependabot[bot] committed:
Bump aiohttp from 3.9.3 to 3.9.4 in /llama-index-integrations/llms/llama-index-llms-llamafile (#12935)

dependabot[bot] committed:
Bump aiohttp from 3.9.3 to 3.9.4 in /llama-index-integrations/agent/llama-index-agent-openai (#12934)

dependabot[bot] committed:
Bump aiohttp from 3.9.3 to 3.9.4 in /llama-index-integrations/tools/llama-index-tools-passio-nutrition-ai (#12933)

dependabot[bot] committed:
Bump aiohttp from 3.9.3 to 3.9.4 in /llama-index-packs/llama-index-packs-koda-retriever (#12932)

dependabot[bot] committed:
Bump aiohttp from 3.9.3 to 3.9.4 in /llama-index-networks/examples/demo/contributor-1 (#12931)

dependabot[bot] committed:
Bump aiohttp from 3.9.3 to 3.9.4 in /llama-index-integrations/llms/llama-index-llms-friendli (#12930)

dependabot[bot] committed:
Bump aiohttp from 3.9.3 to 3.9.4 in /llama-index-networks (#12929)

dependabot[bot] committed:
Bump aiohttp from 3.9.3 to 3.9.4 in /llama-index-networks/examples/demo/contributor-3 (#12928)

dependabot[bot] committed:
Bump aiohttp from 3.9.3 to 3.9.4 in /llama-index-integrations/tools/llama-index-tools-finance (#12927)

dependabot[bot] committed:
Bump aiohttp from 3.9.3 to 3.9.4 in /llama-index-networks/examples/demo/contributor-2 (#12926)

schkovich committed:
Update v0_10_0_migration.md (#12919)

Anush008 committed:
fix: Check for old sparse vector qdrant.base.py (#12916)

tomasonjo committed:
Neo4j metadata filtering support (#12923)

logan-markewich committed:
llama3 cookbook nits (#12952)

logan-markewich committed:
fix dependabot alerts (#12961)

logan-markewich committed:
fix async streaming response from query engine (#12953)

logan-markewich committed:
enforce uuid in element node parsers (#12951)

josem7 committed:
bug(codeHierarchy): fixed bug when function is defined twice (#12941)

nithishr committed:
[docs] Add CouchbaseVectorStore to the Vector Store documentation pages (#12922)

BrandonWeng committed:
Don't retry on all OpenAI APIStatusError exceptions - just InternalServerError (#12947)

philipchung committed:
Fix AsyncOpenAI "RuntimeError: Event loop is closed bug" when instances of AsyncOpenAI are rapidly created & destroyed (#12946)

benjibc committed:
Add new Llama 3 and Mixtral 8x22b model into Llama Index for Fireworks (#12970)

Add new Llama 2 and Mixtral 8x22b model into Llama Index for Fireworks

jerryjliu committed:
add function calling program (#12980)

* cr

* cr

* cr

* cr

* cr

* cr

jerryjliu committed:
update some agents documentation (#12986)

jerryjliu committed:
docs home page nits  (#12987)

jerryjliu committed:
edit chatbot + structured extraction use cases (#12991)

ravi03071991 committed:
Update llama3 cookbook with agents and tools (#13003)

kandros committed:
Update index.md missing verb (#13007)

Data-drone committed:
Fix databricks vector search metadata (#12999)

mattzcarey committed:
Open API Specification Reader (#12998)

regCode committed:
Modify the load_data method to avoid duplicates and reduce the number… (#12977)

UjjwalAyyangar committed:
feat: Adds the index deletion functionality to the WeviateVectoreStore (#12993)

jodua committed:
Updated IBM watsonx foundation models (#12973)

heber-augusto committed:
Avoid infinite loop when not handled exception is raised (#12963)

345ishaan committed:
Return the parent node if the query node is not present (#12983)

armoucar-neon committed:
make the PydanticSingleSelector work with async api (#12964)

jerryjliu committed:
updated QA use cases in the docs (#12989)

jerryjliu committed:
Add llamacloud to docs  (#12982)

* cr

* cr

* cr

---------

Co-authored-by: Logan <logan.markewich@live.com>

jerryjliu committed:
sql docs nit  (#13011)

logan-markewich committed:
import nit for vllm (#13027)

logan-markewich committed:
fix qdrant bug with checking existing collection (#13009)

Javtor committed:
Fix error querying PineconeVectorStore using sparse query mode (#12967)

* fix sparse query pinecone

* none checking

armoucar-neon committed:
Fix: fix query pipeline's arun_with_intermediates (#13002)

logan-markewich committed:
vbump openai (#13026)

nerdai committed:
LATs AgentWorker integration (#13031)

* lats agent worker package

* pants

* add README

* rm lock file

* update example nb

logan-markewich committed:
use collection exists function for qdrant (#13032)

HollowMan6 committed:
Fix class in fusion_retriever notebook (#13034)

logan-markewich committed:
another qdrant fix (#13036)

prajwalsaokar committed:
Integrating AWS DocumentDB as a vector storage method  (#12217)

logan-markewich committed:
missing async call in qdrant (#13039)

drebah committed:
Adding claude 3 opus to BedRock integration (#13033)

* Adding claude 3 opus to BedRock integration

* Increment version

rsrohan99 committed:
Adding support for streaming partial instances of Pydantic output class in OpenAIPydanticProgram (#13021)

* added function to parse partial JSON

* Added stream_partial_objects method in OpenAIPydanticProgram to stream intermediate objects as soon as they're available

* updated documentation to add a section on how to stream intermediate objects using OpenAIPydanticProgram

* version bump for llama-index-program-openai package

* Importing ValidationError from llama_index.core.bridge.pydantic instead

* Moved attribution inside docstring

* Raising ValueError instead of returning None for malformted partial JSON string

* Added unit tests for parse_partial_json function

* removed commented code

ravi03071991 committed:
Add phi-3 benchmarks (#13057)

* Add phi-3 benchmarks

* Update with prompt template

nerdai committed:
Add LLMCompiler Agent Integration (#13044)

nerdai committed:
Add COA Agent integration (#13043)

fzowl committed:
Use voyage-law-2 model in VoyageAI emaple code (#13029)

oandreeva-nv committed:
Improving Docs for NVIDIA TensorRT and Triton integrations (#13060)

logan-markewich committed:
update docs to fix build (#13062)

jerryjliu committed:
docs home page edits  (#13012)

gblazq committed:
Support custom credentials in FirestoreKVStore (#13018)

hemidactylus committed:
(Astra DB integrations) Bump astrapy requirement to at-least-v1 (#12920)

hemidactylus committed:
(Astra db) explicit projection in DB read operations (#12872)

logan-markewich committed:
Tools docs update (#13059)

ptorru committed:
Enabng more OctoAI models - updated model list (#13065)

yisding committed:
fix: qdrant sparse vector backwards compatibility (#13068)

* fix: qdrant sparse vector backwards compatibility

* bump version

alexsherstinsky committed:
[FEATURE] Support both Predibase SDK-v1 and SDK-v2 (#13066)

logan-markewich committed:
v0.10.31 (#13064)

jerryjliu committed:
LATS nits on llamahub  (#13076)

cr

logan-markewich committed:
Qdrant nonetype fix (#13088)

Fix nonetype not iterable

Javtor committed:
Add access control related metadata to SharePoint reader (#13067)

* add access control metadata

* version bump

* add attach_permision flag

brycecf committed:
Corrected wrong output type for `OutputKeys.from_keys()` (#13086)

* Corrected wrong output type for `OutputKeys.from_keys`. Incremented `core` version to `0.10.32`.

* Revert "Corrected wrong output type for `OutputKeys.from_keys`. Incremented `core` version to `0.10.32`."

This reverts commit 1e6391638053ea460f9cc01a2a424efa2a029a75.

* Corrected wrong output type for `OutputKeys.from_keys`.

* Update query.py

---------

Co-authored-by: Andrei Fajardo <92402603+nerdai@users.noreply.github.com>

EmanuelCampos committed:
feat: add run_jobs to aws base embedding (#13096)

0xnoahzhu committed:
feat(core): allow user to customize the keyword extractor prompt temp… (#13083)

Javtor committed:
Add nested filters support for Qdrant and Pinecone (#13113)

nested filters in qdrant and pinecone

RussellLuo committed:
typo fix: Amend addtional_kwargs to additional_kwargs (#13053)

RussellLuo committed:
Do not condense the question if there's no conversation history (#13069)

WoutDeRijck committed:
QueryPlanTool: Execute tool calls in subsequent (dependent) nodes in the query plan (#13047)

Oscilloscope98 committed:
[Community] Support `llama-index-embeddings-ipex-llm` for Intel GPUs  (#13097)

datvodinh committed:
Fix for fusion retriever sometime return Nonetype query(s) before similarity search. (#13112)

logan-markewich committed:
v0.10.32 (#13119)

nataliachodelski committed:
Update llama-index-vector-stores-lantern's index method to 'lantern_hnsw' to maintain compatibility with latest Lantern builds (#13116)

* update lantern postgres access method to 'lantern_hnsw' due to lanterndb removing support for 'hnsw': https://github.com/lanterndata/lantern/commit/c3005aacd9c20df0eaf67fb9b5e1c97711c02518

* bump version number for llama-index-vector-stores-lantern integration from 0.1.2 to 0.1.3

* Added some unit tests for llama-index-vector-stores-lantern (based on existing unit tests in llama-index-vector-stores-postgres)

ZmeiGorynych committed:
Fix kuzu integration .execute() calls, which require a dict not a list of tuples (#13100)

* Fix kuzu execute calls, which require a dict not a list of tuples

* Linter fix

* Increment version in pyproject.toml

* Increment version in the correct pyproject.toml, roll back the other one

Javtor committed:
Use MetadataFilters in LlamaCloud retriever (#13117)

* update llamacloud index

* bump version

logan-markewich committed:
anthropic system prompt (#13131)

logan-markewich committed:
add agent_worker.as_agent() (#13061)

chenrui333 committed:
Update google calendar create_event api to correctly parse start/end_datetime (#13132)

0xnoahzhu committed:
fix(falkordb): skip the create index error when the index is already … (#13085)

* fix(falkordb): skip the create index error when the index is already existing

* chore(falkordb): bump the version to from 0.1.3 to 0.1.4

* fix(integrations+falkordb): specify the error type for the create index error

* fix(integration): revert the legacy update

* chore(integration+falkordb): add an todo comment

* fix(falkordb): add a index existing check before creating index

* Revert "fix(falkordb): add a index existing check before creating index"

This reverts commit 984d33156152af9262ef2cad09cc74d11d138206.

sugamxp committed:
change default value for attendees to empty list (#13134)

fix: change default value for attendees to empty list

fix attendee_list empty issue

Signed-off-by: Rui Chen <rui@chenrui.dev>

gsuriano committed:
Bedrock cohere character limit (#13126)

nerdai committed:
[version] bump version to 0.10.33 (#13138)

clog and vbumps

ragul-kachiappan committed:
Update advanced_retrieval.md - fix link for reranking example notebook (#13153)

logan-markewich committed:
add async methods to ollama and json mode (#13150)

tazarov committed:
chore: Chroma version bump (#13158)

vitali87 committed:
Fix/mymagicai api update (#13148)

logan-markewich committed:
remove error ignoring during streaming (#13160)

logan-markewich committed:
add structured planning agent (#13149)

Rohith-2 committed:
Streaming Support for Nvidia's Triton Integration (#13135)

logan-markewich committed:
update github reader docs (#13181)

mattf committed:
fix direct reranker use in bm25_retriever example (#13171)

yukikwi committed:
Fix: Error when parse file using SimpleFileNodeParser and file's extension doesn't in FILE_NODE_PARSERS (#13156)

Javtor committed:
Exclude access control metadata keys from LLMs and embeddings - SharePoint Reader (#13184)

* add access control metadata

* version bump

* add attach_permision flag

* exclude keys

* version bump

alineberry committed:
add matching source_node.node_id verification (#13109)

* add matching source_node.node_id verification

* check that source_node exists

---------

Co-authored-by: Adam Lineberry <adam.lineberry@kensho.com>

logan-markewich committed:
parameterize queries in pgvector store (#13199)

logan-markewich committed:
add note on text-to-sql usage (#13197)

pr-Mais committed:
Adding Firestore Vector Store (#12048)

logan-markewich committed:
Logan/postgres fix v 2 (#13206)

AgenP committed:
Retrieval Metrics: Updating HitRate and MRR for Evaluation@K documents retrieved. Also adding RR as a separate metric (#12997)

* Updating metrics: MRR renamed to RR, HitRate updated for multi-doc evaluation and new separate MRR implementation

* Updated MRR and HitRate with requested changes

* Iteration w/ class attribute implementation for the calculation option choice

cyril-k committed:
Add tool usage support with text-generation-inference integration from Hugging Face (#12471)

logan-markewich committed:
fix redis kvstore key that was in bytes (#13201)

Troyusrex2 committed:
Added YouTube Metadata Reader (#12975)

mishushakov committed:
feat: Browserbase Web Reader (#12877)

shadyelgewily-slimstock committed:
Add chat summary memory buffer (#13155)

JohnCHarrington committed:
Support passing sqlalchemy.URL for connection string in Postgres (#13123)

EricLBuehler committed:
Integrate mistral.rs LLM (#13105)

* Integrate

* Changes based on comments

* Run pants tailor

* Properly extract and pass logprobs

* Add a simple test

* Add a usage section

* Fix silly mistake

* Add mistralrs as a dependancy

* Fix extract logprobs and update api

* Update for new version

* Update version

* Prettier

* Fix typing

* Remove unnecessary in docs BUILD

anoopshrma committed:
Updating README (#12408)

* updated readme

* more readme updates

* more reader readme updates

* more readme updates

* Update readme for readers done

* linting

---------

Co-authored-by: Logan Markewich <logan.markewich@live.com>

wxywb committed:
Add hybrid retrieval mode to MilvusVectorStore. (#13122)

logan-markewich committed:
update base class for planner agent (#13228)

RosevalJr committed:
[Community] Add streaming for maritalk (#13207)

joelbarmettlerUZH committed:
fix: Respect filter condition for Azure AI Search (#13215)

HairlessVillager committed:
docs: add tips of installing integrations (#13234)

logan-markewich committed:
v0.10.34 (#13235)

* vbump

* fix docs build

nerdai committed:
Add CRITIC agent integration (#13108)

* init

* more scaffolding

* wireframes

* wip nb

* add introspective agent

* add introspective agent

* got memory working better now

* polish

* refactor

* refactor

* revert core changes

* rm parent_task_id

* refactor for better org

* clean up critic class

* support async for critic

* carry step state forward to next steps

* add dispatcher

* make max_iterations public

* add stopping callable

* implement jerry self reflection

* bug fix and async support for self reflection

* proper verbose propagation

* add mini showdown

* fix tests and llamahub info

* clean up IntrospectiveAgentWorker

* clean up SelfReflection

* update README

* update README

* finish README

* readme wip

* readme wip

* readme wip

* update nb and add copy to docs

* rm dev nb

* remove lock file

* use structured_predict

* update nb

* line up chat history for introspective agent better

* update nb

* update nbs

nerdai committed:
chore: fix link to example notebook for introspective agent (#13236)

logan-markewich committed:
fix import for web readers (#13248)

Commit:
fix: AzStorageBlobReader (#13246) (#13247)

logan-markewich committed:
fix readme links (#13249)

hassiebp committed:
fix(docs): langfuse API reference page (#13242)

ignacioct committed:
Argilla Callback Handler README updated (#13245)

yoonsch217 committed:
Fix log message for retrieved nodes count in chromadb (#13238)

logan-markewich committed:
add mistralrs example to docs (#13250)

mattf committed:
add nvidia nim rerank support (#13178)

Commit:
Add support for Vertex AI Vector Search as Vector Store (#13186)

mattf committed:
add nvidia nim embeddings support (#13177)

JoanFM committed:
feat: add encoding_type parameters in JinaEmbedding class (#13172)

tianzedavid committed:
chore: remove repetitive words (#13169)

nerdai committed:
FIX: `is_done` undefined error with ToolInteractiveReflectionAgent + Unit tests (#13252)

* nit

* add self reflection test

* fix is_done bug add unit test tool interactive

* add test_introspective.py

* bump version number

dependabot[bot] committed:
Bump tqdm from 4.66.2 to 4.66.3 in /llama-index-packs/llama-index-packs-diff-private-simple-dataset (#13261)

Bump tqdm

Bumps [tqdm](https://github.com/tqdm/tqdm) from 4.66.2 to 4.66.3.
- [Release notes](https://github.com/tqdm/tqdm/releases)
- [Commits](https://github.com/tqdm/tqdm/compare/v4.66.2...v4.66.3)

---
updated-dependencies:
- dependency-name: tqdm
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

SinghCoder committed:
chore: remove errors from example (#13270)

* remove errors from example

* Update embeddings.md

---------

Co-authored-by: Andrei Fajardo <92402603+nerdai@users.noreply.github.com>

dependabot[bot] committed:
Bump tqdm from 4.66.2 to 4.66.3 in /llama-index-integrations/tools/llama-index-tools-finance (#13260)

dependabot[bot] committed:
Bump tqdm from 4.66.2 to 4.66.3 in /llama-index-integrations/llms/llama-index-llms-friendli (#13258)

dependabot[bot] committed:
Bump tqdm from 4.66.2 to 4.66.3 in /llama-index-packs/llama-index-packs-koda-retriever (#13257)

dependabot[bot] committed:
Bump tqdm from 4.66.2 to 4.66.3 in /llama-index-networks (#13256)

dependabot[bot] committed:
Bump tqdm from 4.66.2 to 4.66.3 in /llama-index-core (#13255)

RajeshThallam committed:
[docs] Add Vertex AI Vector Search to the Vector Store documentation … (#13254)

rozerarenu committed:
Fix bedrock notebook retrieve call and add llamahub doc improvement (#13262)

thomasht86 committed:
Add VectorStore integration for Vespa  (#13213)

sandangel committed:
fix: azure openai api key not picking up env var (#13267)

busaud committed:
docstring typo (#13273)

Maybe this was copied from chain of thought implementation.

RussellLuo committed:
Fix the import of Pydantic (#13275)

Thomas-AH-Heller committed:
Fix async issue when streaming with Mistral AI (#13292)

zhoukuncheng committed:
Fix incorrect link in the docs/use_cases/agents.md (#13285)

dependabot[bot] committed:
Bump jinja2 from 3.1.3 to 3.1.4 in /docs (#13293)

dependabot[bot] committed:
Bump jinja2 from 3.1.3 to 3.1.4 in /llama-index-packs/llama-index-packs-diff-private-simple-dataset (#13297)

dependabot[bot] committed:
Bump jinja2 from 3.1.3 to 3.1.4 in /llama-index-networks (#13296)

dependabot[bot] committed:
Bump jinja2 from 3.1.3 to 3.1.4 in /llama-index-packs/llama-index-packs-koda-retriever (#13295)

dependabot[bot] committed:
Bump jinja2 from 3.1.3 to 3.1.4 (#13294)

RussellLuo committed:
Handle llm properly for QASummaryQueryEngineBuilder and RouterQueryEngine (#13281)

sravan-onefc committed:
Fix Bug in sql join statement (#13277)

dependabot[bot] committed:
Bump jinja2 from 3.1.3 to 3.1.4 in /llama-index-core (#13298)

wxywb committed:
Update Milvus hybrid retrieval information in related documentations. (#13282)

Signed-off-by: wxywb <xy.wang@zilliz.com>

WilliamEspegren committed:
feat: Spider Web Loader (#13200)

* files

* spider integration

* pre-commit

* bump v, change init

* Documentation

* lint & format

* metadata defualt set in params

* metadata set in param

* lint

SilasMarvin committed:
Fixed spelling (#13300)

WilliamEspegren committed:
Bump version (#13301)

dependabot[bot] committed:
Bump jinja2 from 3.1.3 to 3.1.4 in /llama-index-integrations/llms/llama-index-llms-friendli (#13304)

dependabot[bot] committed:
Bump jinja2 from 3.1.3 to 3.1.4 in /llama-index-integrations/tools/llama-index-tools-finance (#13303)

logan-markewich committed:
expand instrumentation payloads (#13302)
